src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -76,15 +79,23 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
             _logger.Started();
         }
 
-        private async Task HandleBackgroundException(BackgroundService backgroundService)
+        private async Task TryStartBackgroundServiceAsync(
+            BackgroundService backgroundService, CancellationToken cancellationToken)
         {
             try
             {
+                // Fire IHostedService.Start
+                await backgroundService.StartAsync(cancellationToken).ConfigureAwait(false);
                 await backgroundService.ExecuteTask.ConfigureAwait(false);
             }
             catch (Exception ex)
             {
                 _logger.BackgroundServiceFaulted(ex);
+                if (_options is { BackgroundServiceExceptionBehavior: BackgroundServiceExceptionBehavior.StopHost })
+                {
+                    _logger.LogCritical(SR.Format(SR.BackgroundServiceExceptionStoppedHost, ex));
```

Comment:
I would move this to `HostingLoggerExtensions` like `BackgroundServiceFaulted` is. And then define an eventId.

Also, the Exception should be passed to `LogCritical`, so the logger can do whatever it wants with it. You shouldn't be formatting it into the message yourself.

Reply:
Done